core/lib/assert.c

```diff
@@ -19,16 +19,25 @@
 #include "architecture.h"
 #include "cpu.h"
 #include "panic.h"
+#if IS_USED(MODULE_BACKTRACE)
+#include "backtrace.h"
+#endif
 
 __NORETURN void _assert_failure(const char *file, unsigned line)
 {
     printf("%s:%u => ", file, line);
+#if IS_USED(MODULE_BACKTRACE)
+    backtrace_print();
+#endif
```

Comment:
As there is no line feed in the line above, the alternative code path should add one (but keep the old format in check, if it is not included)

```suggestion
#if IS_USED(MODULE_BACKTRACE)
    printf("failed assertion. Backtrace:\n"
    backtrace_print();
#endif
```